src/lib.rs

```diff
@@ -33,30 +33,46 @@
     variant_size_differences
 )]
 
+#[cfg(feature = "mock")]
+#[macro_use]
+extern crate lazy_static;
 extern crate maidsafe_utilities;
+#[cfg(any(test, feature = "mock"))]
 extern crate rand;
-#[cfg(not(feature = "use-mock-crypto"))]
+#[cfg(not(feature = "mock"))]
 extern crate rust_sodium;
 extern crate serde;
 #[macro_use]
 extern crate serde_derive;
+#[cfg(not(feature = "mock"))]
+extern crate tiny_keccak;
 #[macro_use]
 extern crate quick_error;
-#[cfg(test)]
+#[cfg(any(test, feature = "mock"))]
 #[macro_use]
 extern crate unwrap;
 
-#[cfg(feature = "use-mock-crypto")]
+#[cfg(feature = "mock")]
 mod mock_crypto;
-#[cfg(feature = "use-mock-crypto")]
+#[cfg(feature = "mock")]
+mod seeded_rng;
+#[cfg(feature = "mock")]
 use mock_crypto::rust_sodium;
+#[cfg(feature = "mock")]
+use mock_crypto::tiny_keccak;
+#[cfg(feature = "mock")]
+pub use seeded_rng::SeededRng;
 
 use maidsafe_utilities::serialisation::{deserialise, serialise, SerialisationError};
+#[cfg(feature = "mock")]
 use rand::Rng;
 use rust_sodium::crypto::{box_, sealedbox, secretbox, sign};
 use serde::{de::DeserializeOwned, Serialize};
 use std::sync::Arc;
 
+/// Represents public signature key.
+pub type PublicSignKey = [u8; 32];
```

Comment:
Did we agree to export something like `const PUBLIC_SIGN_KEY_BYTES: usize = 32;` and use that here?

Reply:
I didn't write it in my notes as getting agreed, rather as point was raised but it's Routing that wants to set this limit and not the other way round - so if it was configurable then yes it's helpful but if not, the constant is basically Routing's and this is just here as a (in absence of a better way) a reverse dependency. Maybe instead of the constant we could wait for future versions of Rust to make constants configurable by user-libs and then make this change. It's not a big point so I'm OK if we want a constant just now too.